app/src/main/java/org/listenbrainz/android/ui/screens/brainzplayer/BrainzPlayerBackDropScreen.kt

```diff
@@ -206,7 +208,19 @@ fun PlayerScreen(
     } else {
         println("Playlist is empty")
     }
-
+    //For handling song change by list or buttons
+    LaunchedEffect(brainzPlayerViewModel.appPreferences.currentPlayable?.currentSongIndex) {
+        pagerState.scrollToPage(
+            brainzPlayerViewModel.appPreferences.currentPlayable?.currentSongIndex ?: 0
+        )
+    }
+    //For handling song change by pager
+    LaunchedEffect(pagerState) {
```

Comment:
Instead of collecting snapshot flow, just do `LaunchedEffect(pagerState.currentPage)`